modules/lib/src/main/resources/assets/js/v6/features/views/context/widget/dependencies/index.tsx

```diff
@@ -0,0 +1,134 @@
+import {ReactElement, useEffect, useState} from 'react';
+import {LegacyElement} from '../../../../shared/LegacyElement';
+import {ContentSummaryAndCompareStatus} from '../../../../../../app/content/ContentSummaryAndCompareStatus';
+import {WidgetItemViewInterface} from '../../../../../../app/view/context/WidgetItemView';
+import {$contextContent} from '../../../../store/context/contextContent.store';
+import {useStore} from '@nanostores/preact';
+import {ResolveDependenciesRequest} from '../../../../../../app/resource/ResolveDependenciesRequest';
+import {DependencyType} from '../../../../../../app/browse/DependencyType';
+import {ContentTypeName} from '@enonic/lib-admin-ui/schema/content/ContentTypeName';
+import {ContentDependencyJson} from '../../../../../../app/resource/json/ContentDependencyJson';
+import {DependenciesWidgetContentSection} from './DependenciesWidgetContentSection';
+import {DependenciesWidgetFlowSection} from './DependenciesWidgetFlowSection';
+import Q from 'q';
+
+export type DependencyItem = {
+    type: DependencyType;
+    itemCount: number;
+    iconUrl: string;
+    contentType: ContentTypeName;
+};
+
+const DEPENDENCIES_WIDGET_NAME = 'DependenciesWidget';
+
+const DependenciesWidget = (): ReactElement => {
+    const content = useStore($contextContent);
+    const [inboundDependencies, setInboundDependencies] = useState<DependencyItem[]>([]);
+    const [outboundDependencies, setOutboundDependencies] = useState<DependencyItem[]>([]);
+
+    useEffect(() => {
+        if (!content) {
+            setInboundDependencies([]);
+            setOutboundDependencies([]);
+            return;
+        }
+
+        resolveDependencies(content).then((dependencyEntry) => {
+            const inbound = dependencyEntry.inbound.map((item) => ({
+                type: DependencyType.INBOUND,
+                itemCount: item.count,
+                iconUrl: item.iconUrl,
+                contentType: new ContentTypeName(item.type),
+            }));
+
+            setInboundDependencies(inbound);
+
+            const outbound = dependencyEntry.outbound.map((item) => ({
+                type: DependencyType.OUTBOUND,
+                itemCount: item.count,
+                iconUrl: item.iconUrl,
+                contentType: new ContentTypeName(item.type),
+            }));
+
+            setOutboundDependencies(outbound);
+        });
+    }, [content]);
+
+    if (!content) return undefined;
```

Comment:
Return `null`. Won't change anything, but `null` is used by React by convention as absence of component or property.